client/state/selectors/get-theme-install-url.ts

```diff
@@ -0,0 +1,11 @@
+import { getSiteAdminUrl, getSiteSlug, isAdminInterfaceWPAdmin } from '../sites/selectors';
+import type { AppState } from 'calypso/types';
+
+export default function getThemeInstallUrl( state: AppState, siteId?: number | null ) {
+	if ( ! isAdminInterfaceWPAdmin( state, siteId ) ) {
+		const siteSlug = getSiteSlug( state, siteId );
+		return `/themes/${ siteSlug }`;
+	}
+
+	return getSiteAdminUrl( state, siteId, 'theme-install.php' );
```

Comment:
Not sure if it's intentional to disregard this pattern, but just putting it out there that Calypso no longer encourages linking to the Themes page in WP-Admin elsewhere (even though it's still directly accessible!) - might be worth extending that here too and forcing the Calypso page? :)

From #84715:

> The wp-admin Theme Showcase has several issues that we won't fix in the short term. Its filtering hasn't been working in a while; it displays just a fraction of our themes; there is no editorial curation; it still shows individual Premium theme prices even though we haven't supported individual Premium theme purchases in a long time.

Reply:
It is intentional! We are starting to promote (and fix!) wp-admin screens, stay tuned! 😄